kitsune/search/v2/search.py

```diff
@@ -0,0 +1,238 @@
+
+    def make_result(self, hit, locale):
+        summary = first_highlight(hit)
+        if not summary:
+            summary = hit.question_content[locale][:500]
+        summary = sanitize(summary)
+
+        num_answers = 0
+        answer_content = getattr(hit, "answer_content", None)
+        if answer_content:
+            num_answers = len(answer_content[locale])
+
+        return {
+            "type": "question",
+            "url": "/questions/{}".format(hit.question_id),
+            "score": hit.meta.score,
+            "title": hit.question_title[locale],
+            "search_summary": summary,
+            "last_updated": datetime.fromisoformat(hit.question_updated),
+            "is_solved": hit.question_has_solution,
+            "num_answers": num_answers,
+            "num_votes": hit.question_num_votes,
+        }
+
+
+class KBConfig(BaseConfig):
+    """Search config for the Knowledge Base."""
+
+    index = WIKI_DOCUMENT_INDEX_NAME
+    fields = [
+        "keywords.{}^8",
+        "title.{}^6",
+        "summary.{}^2",
+        "content.{}^1",
+    ]
+    highlight_fields = ["summary.{}", "content.{}"]
+
+    @property
+    def filter(self):
+        return Q(
+            "bool",
+            filter=super().filter,
+            must_not=Q("terms", category=[TEMPLATES_CATEGORY, CANNED_RESPONSES_CATEGORY]),
+        )
+
+    def product_filter(self, product):
+        return Q("term", product_ids=product.id)
+
+    def make_result(self, hit, locale):
+        summary = first_highlight(hit)
+        if not summary and hasattr(hit, "summary"):
+            summary = getattr(hit.summary, locale, None)
+        if not summary:
+            summary = hit.content[locale][:500]
+        summary = sanitize(summary)
+
+        return {
+            "type": "document",
+            "url": "/{}/kb/{}".format(locale, hit.slug[locale]),
+            "score": hit.meta.score,
+            "title": hit.title[locale],
+            "search_summary": summary,
+        }
+
+
+class Search:
+    """Create an ES7 Search."""
+
+    def __init__(self, query, language, product=None, page=1):
+        self.config_classes = []
+        self.query = query
+        self.language = language
+        self.product = product
+        self.page = page
+
+    def add_config(self, config):
+        self.config_classes.append(config())
+
+    def execute(self):
+        # determine indices to search over, and create DSL search object
+        index = ",".join(self._getattr("index"))
+        search = DSLSearch(using=es7_client(), index=index)
+
+        # add filters
+        filters = self._getattr("filter")
+        search = search.query("bool", should=filters)
+
+        # add product filters
+        if self.product:
+            product_filters = self._getattr("product_filter", self.product)
+            search = search.query("bool", should=product_filters)
+
+        # add query
+        fields = self._localize_fields(self._getattr("fields"))
```

Comment:
this does not localize any field but rather retrieving the localized version of it

Reply:
this is clean but it took me a moment to figure out what `_getattr('fields')` does here. I am not particularly suggesting something here - just a general remark